src/Microsoft.DotNet.Build.Tasks/Targets/versioning.targets

```diff
@@ -41,6 +41,7 @@
       Outputs="$(AssemblyInfoFile)"
       Condition="'$(GenerateAssemblyInfo)'=='true'">
 
+    <MakeDir Condition="!Exists('$(AssemblyInfoFileDir)')" Directories="$(AssemblyInfoFileDir)" ContinueOnError="true" />
```

Comment:
I believe MakeDir already handles if the dir already exists, so I don't believe the condition is necessary.